### PR TITLE
Revert "Extend `Eigen` to keep additional information from `geevx` (#…

### PR DESCRIPTION
**Pull request by andreasnoack**
…38483)"

This reverts commit 9d3a7c47df972b1108b5a49d2c5ce493635bbadf.

See https://github.com/JuliaLang/LinearAlgebra.jl/issues/859 for the issues. Instead, we should work on a new version of https://github.com/JuliaLang/julia/pull/38483 for the 1.8 release.

### CODE REVIEW

**Comment by andreasnoack** [comment]
I've added an extra commit since https://github.com/JuliaLang/julia/pull/41352 was using the reverted functionality.

File: stdlib/LinearAlgebra/test/eigen.jl [hunk 1]
```diff
@@ -189,24 +189,18 @@ end
     @test B isa Eigen{Float16, Float16, Matrix{Float16}, Vector{Float16}, Vector{Float16}}
```
Comment (dkarrasch) [inline stdlib/LinearAlgebra/test/eigen.jl:189]:
```suggestion
    @test B isa Eigen{Float16, Float16, Matrix{Float16}, Vector{Float16}}
```

File: stdlib/LinearAlgebra/test/eigen.jl [hunk 2]
```diff
@@ -189,24 +189,18 @@ end
...
     @test B.values isa Vector{Float16}
     @test B.vectors isa Matrix{Float16}
-    @test B.vectorsl isa Matrix{Float16}
     @test B.values ≈ B32.values
     @test B.vectors ≈ B32.vectors
-    @test B.vectorsl ≈ B32.vectorsl
     @test D isa Eigen{ComplexF16, ComplexF16, Matrix{ComplexF16}, Vector{ComplexF16}, Vector{Float16}}
```
Comment (dkarrasch) [inline stdlib/LinearAlgebra/test/eigen.jl:194]:
```suggestion
    @test D isa Eigen{ComplexF16, ComplexF16, Matrix{ComplexF16}, Vector{ComplexF16}}
```

File: stdlib/LinearAlgebra/test/eigen.jl [hunk 3]
```diff
@@ -189,24 +189,18 @@ end
...
     @test D.values isa Vector{ComplexF16}
     @test D.vectors isa Matrix{ComplexF16}
-    @test D.vectorsl isa Matrix{ComplexF16}
     @test D.values ≈ D32.values
     @test D.vectors ≈ D32.vectors
-    @test D.vectorsl ≈ D32.vectorsl
     @test F isa Eigen{ComplexF16, ComplexF16, Matrix{ComplexF16}, Vector{ComplexF16}, Vector{Float16}}
```
Comment (dkarrasch) [inline stdlib/LinearAlgebra/test/eigen.jl:199]:
```suggestion
    @test F isa Eigen{ComplexF16, ComplexF16, Matrix{ComplexF16}, Vector{ComplexF16}}
```

**Review by dkarrasch — COMMENTED** [review]
These appear to have too many type parameters (tests from the Float16 PR).

**Pull request merged**